### PR TITLE
Conversion of exp.txt to Libconfig

### PR DESCRIPTION
**Pull request by Smokexyz**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Conversion of exp.txt to libconfig.
Addition of new file `db/<re/pre-re>/exp_group_db.conf` listing common exp groups.
Groups are linked with classes through key-value pairs in the job_db.conf.

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** master

[//]: # (Master? Slave?)

**Issues addressed:**

This conversion also removes unnecessary, unclear and bulky code from legacy eAthena. 
```
Memory consumption before: (247 * 8 * 2 * max_lv) bytes. (lv 175 = 691,600 bytes)
Cause: 2 fixed length arrays - uint64_t exp_table[CLASS_COUNT][2][MAX_LEVEL] and uint32_t max_level[CLASS_COUNT][2];
```
```
Memory consumption now: (36 + (8 * max_lv)) * 2 bytes. (lv 175 <= 2,872 bytes).
Cause: Single in-memory vector of exp groups that are referenced from a table of classes.
```

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List

- [ ] Test more thoroughly.

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1944)

File: src/common/mmo.h [hunk 1]
```diff
@@ -1090,6 +1090,15 @@ enum e_class {
 #endif
 };
 
+#define job_is_dummy(job) \
```
Comment (4144) [inline src/common/mmo.h]: probably this better implement as function.
Because some one may want change this list from plugin.

File: src/map/pc.c [hunk 2]
```diff
@@ -12030,14 +12008,25 @@ void pc_update_job_and_level(struct map_session_data *sd)
...
+
 	db_destroy(pc->itemcd_db);
 	pc->at_db->destroy(pc->at_db,pc->autotrade_final);
 
 	pcg->final();
 
 	pc->clear_skill_tree();
 
+
+	for (k = 0; k < 2; k++) {
```
Comment (4144) [inline src/map/pc.c]: this probably better move to separate function. Something like pc_clear_exp_groups.

**Comment by EyesOfAHawk** [comment]
great stuff. hopefully all the old .txt files from eAthena days can one day be converted (abra_db, sc_config, mob_skill_db, etc.)

**Pull request closed without merging**